File: codegen/gen-plugins.py
import json
from pathlib import Path
from operator import itemgetter

PLUGINS_JSON: Path = Path(__file__).parents[1] / 'plugins.json'
extract_plugin = itemgetter('gh')

TEMPLATE = \
'''return {{
{0}
}}
'''

def main() -> None:
    with open(PLUGINS_JSON, 'r', encoding='utf8') as f:
        plugins_json: dict = json.load(f)

    plugin_names = map(extract_plugin, plugins_json.values())
    lazy_fmt = map(lambda p: f'    {{ "{p}" }},', plugin_names)
    plugin_list = '\n'.join(list(lazy_fmt))


    print(TEMPLATE.format(plugin_list), end='')

if __name__ == '__main__':
    main()

File: codegen/gen-plugins.pl
use strict;
use warnings;
use diagnostics;

use v5.16;

use JSON::PP;
use File::Basename;
use File::Spec;

my $PROJECT_ROOT = dirname(dirname(__FILE__));
my $PLUGINS_JSON = File::Spec->catfile($PROJECT_ROOT, 'plugins.json');

open(my $fh, '<', $PLUGINS_JSON) or die "Cannot open file: $!";
my $json_text = do { local $/; <$fh> };
close($fh);

my $plugins_json = decode_json($json_text);
my @plugin_names = map { $_->{gh} } values %$plugins_json;
my @lazy_fmt = map { "    { \"$_\" }," } @plugin_names;
@lazy_fmt = sort @lazy_fmt;

my $plug_file = "return {\n" . join("\n", @lazy_fmt) . "\n}";

print "$plug_file\n";
